test/controllers/users/sessions_controller_test.rb

```diff
@@ -124,4 +124,111 @@ class Users::SessionsControllerTest < ActionDispatch::IntegrationTest
+      assert_redirected_to root_url(subdomain: @restarone_subdomain.name)
+
+      # It should render private cms-page successfully if the user can view restricted pages
+      @restarone_user.update!(can_view_restricted_pages: true)
+      sign_in(@restarone_user)
+      get comfy_cms_render_page_url(subdomain: @restarone_subdomain.name, cms_path: 'test-cms-page')
+      assert :success
+    end
+  end
+
+  test 'redirects to private forum-thread page after sign-in successfully' do
+    @restarone_subdomain.update!(forum_is_private: true)
+
+    Apartment::Tenant.switch @restarone_subdomain.name do
+      @restarone_user.update!(can_access_admin: true)
+
+      forum_category = ForumCategory.create!(name: 'test', slug: 'test')
+      forum_thread = @restarone_user.forum_threads.create!(title: 'Test Thread 1', forum_category_id: forum_category.id)
+  
+      # First, trying to visit a private forum-thread page which redirects to sign_in page
+      get simple_discussion.forum_thread_url(subdomain: @restarone_subdomain.name, id: forum_thread.id)
+      assert_redirected_to new_user_session_url(subdomain: @restarone_subdomain.name)
+  
+      payload = {
+        user: {
+          email: @restarone_user.email,
+          password: '123456'
+        }
+      }
+
+      # After signing-in, it should redirect back to the private forum-thread page
+      post user_session_url(subdomain: @restarone_subdomain.name), params: payload
+      assert_redirected_to simple_discussion.forum_thread_url(subdomain: @restarone_subdomain.name, id: forum_thread.id)
+    end
+  end
+
+  test 'redirects to api-namespace page after sign-in successfully' do
+    Apartment::Tenant.switch @restarone_subdomain.name do
+      @restarone_user.update!(can_access_admin: true)
+
+      api_namespace = ApiNamespace.create!(name: 'clients',
+        slug: 'clients',
+        version: 1,
+        properties: {'name': 'test'},
+        requires_authentication: false,
+        namespace_type: 'create-read-update-delete')
+
+      # First, trying to visit api-namespace page which redirects to sign_in page
+      get api_namespace_url(subdomain: @restarone_subdomain.name, id: api_namespace.id)
+      assert_redirected_to new_user_session_url(subdomain: @restarone_subdomain.name)
+  
+      payload = {
+        user: {
+          email: @restarone_user.email,
+          password: '123456'
+        }
+      }
+
+      # After signing-in, it should redirect back to the api-namespace page
+      post user_session_url(subdomain: @restarone_subdomain.name), params: payload
+      assert_redirected_to api_namespace_url(subdomain: @restarone_subdomain.name, id: api_namespace.id)
+
+      # The response will be success if the user can manage api
+      @restarone_user.update!(can_manage_api: true)
+      sign_in(@restarone_user)
+      get api_namespace_url(subdomain: @restarone_subdomain.name, id: api_namespace.id)
+      assert :success
```

Comment:
@alis-khadka shouldn't this be `assert_response :success` ? Im not sure what `assert :success` does, I don't see any docs on it eg: https://apidock.com/rails/ActionController/Assertions/ResponseAssertions/assert_response

If they are equivalent, happy to approve as is